vectors.lisp

```diff
@@ -65,6 +65,7 @@ This is to vectors what `values-list' is to lists."
     `(trivia:guard (and ,it (trivia:vector* ,@elts))
                    (= (length ,it) ,(length elts)))))
 
+(declaim (notinline pad-start))
```

Comment:
Declaiming a function notinline means that its compiler macro will never be used, which would defeat the purpose here (and for pad-end).

Reply:
That was not on my radar. 
Now the code is restructured which resolves these style-warnings without notinline declarations:
[Fix style-warning: compiled before compiler-macro](https://github.com/ruricolist/serapeum/pull/133/commits/c0d494e55e51cd4c8ffb21dc4692babc2c2b3ed1)